### PR TITLE
refactor: Add ChainClient setMockTime, getWallets methods

### PR DESCRIPTION
**Pull request by ryanofsky**
This PR is part of the [process separation project](https://github.com/bitcoin/bitcoin/projects/10).

These changes are needed to set mock times, and get wallet interface pointers correctly when
wallet code is running in a different process from node code in #10102

### CODE REVIEW

**Comment by MarcoFalke** [comment]
re-run ci

File: src/interfaces/chain.h [hunk 1]
```diff
@@ -280,6 +282,12 @@ class ChainClient
 
     //! Shut down client.
     virtual void stop() = 0;
+
+    //! Set mock time.
+    virtual void setMockTime(int64_t time) = 0;
+
+    //! Return interfaces for accessing wallets (if any).
+    virtual std::vector<std::unique_ptr<Wallet>> getWallets() { return {}; }
```
Comment (promag) [inline src/interfaces/chain.h]: Any reason to have definition?

**Review by promag — COMMENTED** [review]
Code review ACK, but could squash and avoid the temporary circular dependency.

File: src/interfaces/chain.h (hunk 1 above)
Comment (MarcoFalke) [reply src/interfaces/chain.h]: I presume this is needed when there is more than one chain client that has wallets. Maybe the doc string could be clarified to say "of the wallets of all chain clients" or similar.

**Review by maflcko — APPROVED** [review]
ACK 06edeb442971747634d1558130d15e6227325422 🚻

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 06edeb442971747634d1558130d15e6227325422 🚻
-----BEGIN PGP SIGNATURE-----

iQGyBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhEFQvzBP27AqgANDYm0anOBCrcFhiEuALQa0nLCp7dkxK34mOqJEWsfzYBjKL8
kdPgbcyOMx9cc0N1YzQJuVUgFsDEzQjv4+SSogMmec00FEIUPl6rJpVMnmeMbsju
+5nmilRKRgZ0TR1Z2dAebe84Cm8go5Il6Il1vI0WcsIE4RzrNVMFAqaVAI/RV+4g
EpOzYCwzEgXMGWw4nixku/hrIVRbAlOTsKrOncYmp10/BtDvT4EhJq7D1w1q67KV
HDgd5PVa8geRmy7DTEvEjpQEYnL2qBR2yr1WtMwzxWBSGrwqSLCfl7RyQuWgX/2x
oMnQzoCszSO9gQM1X7BIag8io0H+hrTgo4OTkWmKxwJHuUX3A8JXTksAPyvw5CSR
CM2bxOZUu6hmdrTCrMWHZTbuexn2AQsEFicIWzlP3Kbi/FQXSC7dmx013KHUSmEj
K1Fg9FZCBvCJq6nO7Eg4MK5qhYDFzWmTkxPWa10zVNc3/Cto4AWPIJnl9AjjkX8o
u+ofy1Y=
=OnjJ
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `7dc65ddc41d7a97c1cc5b926e0f20c04316b6c5b18b17537c4fc59c620d2d263  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401087dc65ddc41d7a97c1cc5b926e0f20c04316b6c5b18b17537c4fc59c620d2d263f0108401f1dfe64a04c75e546c724b68d8d608fff0103bae0bbb8f2b67a539ad03bc7850c47508f1045e33203df00860ac11a96026650f0083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff0103c383fe5faa20a0030a338fd171aad0308f1045e33203df008917834943306bab20083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010567fe534589dc008b155af82ca04e5b908f1045e33203cf008f5fa000323eaccce0083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010a2685bb35e908bb3cd78f9456618719808f1045e33203cf008bdc72c0a8c527a7e0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Review by ryanofsky — COMMENTED** [review]
Rebased 06edeb442971747634d1558130d15e6227325422 -> 3ce16ad2f91d1e2edc7e7bdc5a19f72aa8c3e739 ([`pr/ipc-clients.1`](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-clients.1) -> [`pr/ipc-clients.2`](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-clients.2), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/ipc-clients.1-rebase..pr/ipc-clients.2)) due to conflict with #18037 and removed extraneous include and method definition.

**Review by promag — COMMENTED** [review]
Code review ACK 3ce16ad2f91d1e2edc7e7bdc5a19f72aa8c3e739.

**Comment by MarcoFalke** [comment]
re-ACK 3ce16ad2f91d1e2edc7e7bdc5a19f72aa8c3e739 🔙

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

re-ACK 3ce16ad2f91d1e2edc7e7bdc5a19f72aa8c3e739 🔙
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUguhwv+OeT+gPBC/ph0lphxilHyBJqekSTWMBNvVtlusfJujyxbuCxlv0okK1IS
3lFhGkeEfEj2/s2H8RMsvDVSjK5BN4/06SHYb99I9POCyjqkeg+grYfUjcqVnNmq
U5pPpngZln9OpRd3AMdU7caw8uXcX4l5wak/tYVZRE9QKybF3xPlpHzcpRkdVYnv
ZKXSwaGW5gcsUMstlUzzHq1J59XZAk5hHrH3tvi/JcOZPQFD6McKStlTXDXAERBa
Fp4cA4K+dn9HkJegMCDw4BSp1bt9oVJXvQshargnDfaYtRQgFDq55O/rtSVJYF4v
Ey9o2l8tzbVGz9VABXueSC0mDRz/cN8raO/V50O1guzK5tNlDGJuvRU9VrXWgyR+
xPeStH4sJQLzSiuy6aX/7FNIqkPPHk88l14KWLNPyv+T6+YyXWQdTDQF73DlkRQq
y2oZ5h0zmOdBZTlwbD21hVQpejDcyFYI9AIihfJzDWjPgN2NWl6ia57Uh08hCX+z
M1cZESy9
=Ly4W
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `d3df18e203f28d9eaeee0acf457efa8312dc6a5a45c7a803a6b0ac74106c450d  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108d3df18e203f28d9eaeee0acf457efa8312dc6a5a45c7a803a6b0ac74106c450df0106e6f4d2717ba690ed8aa7d1d732451a808fff0105efb70f8d1ce411198fed962a774860808f1045e5686f0f008094a98164ed238af0083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010c7e00732f3a1ea7e8bf1cda401ce213608f1045e5686f0f0081f351fd2c56121be0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010ec120f4c6c3f4f023020f444f795039c08f1045e5686f0f008a9746284ade9dde50083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010f028fe2169223be22ecbce77c31ca3de08f02016b30270f71609722822bb82efe3e18575c015712f73d5153c0fccaf82c78ad808f1045e5686f2f008d0344a6711e5b2780083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6d)

</details>

**Pull request merged**